flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -248,35 +248,34 @@ arrayLoadExtValue(fir::FirOpBuilder &builder, mlir::Location loc,
     }
     arrTy = ty.cast<fir::SequenceType>();
   }
-  // Recycle componentToExtendedValue if it looks plausible.
-  if (!fir::hasDynamicSize(arrTy))
-    return fir::factory::componentToExtendedValue(builder, loc, newBase);
-
-  auto eleTy = fir::unwrapSequenceType(arrTy);
-  if (!load.shape()) {
-    // ???: The final argument is a BoxValue, but that's what we are trying to
-    // recover here.
-    auto exv = fir::factory::readBoxValue(builder, loc, load.memref());
-    return fir::substBase(exv, newBase);
-  }
-  auto extents = fir::factory::getExtents(load.shape());
-  auto lbounds = fir::factory::getOrigins(load.shape());
-  if (fir::isa_char(eleTy)) {
-    auto len = newLen;
-    if (!len)
-      len = fir::factory::CharacterExprHelper{builder, loc}.getLength(
-          load.memref());
-    if (!len) {
-      assert(load.typeparams().size() == 1 && "length must be in array_load");
-      len = load.typeparams()[0];
+
+  // Use the shape op, if there is one.
+  if (mlir::Value shapeVal = load.shape()) {
+    mlir::Type eleTy = fir::unwrapSequenceType(arrTy);
+    std::vector<mlir::Value> extents = fir::factory::getExtents(shapeVal);
```

Comment:
What happens if the shapeVal is a ShiftOp with no extent information because the base is a fir.box ?